packages/web3Connector/index.js

```diff
@@ -0,0 +1,54 @@
+const path = require('path');
+const fs = require('fs');
+
+module.exports = (embark) => {
+  embark.events.on('runcode:ready', () => {
```

Comment:
Maybe you can promisify the events:
```
function whenRuncodeReady(embark) {
  return new Promise(resolve) {
    embark.events.on('runcode:ready', () => {
      resolve();
    })
  }
}
```

That way you can flatten all this code:
`
await whenRuncodeReady()
...
`

And apply the same technique to all below